### ISSUE TITLE
Auto unsubscribe issue: Hello, world?)

### ISSUE THREAD

**Issue by vellrya** [nats-io/nats.py#279]
Hello. I haven't used NATS in Python before, and I wanted to try the example from the Readme. I only changed the server address and ran it like this:

```
import asyncio
import nats
from nats.errors import ConnectionClosedError, TimeoutError, NoServersError

async def main():
    # It is very likely that the demo server will see traffic from clients other than yours.
    # To avoid this, start your own locally and modify the example to use it.
    nc = await nats.connect("nats://localhost:4222")

    # You can also use the following for TLS against the demo server.
    #
    # nc = await nats.connect("tls://demo.nats.io:4443")

    async def message_handler(msg):
        subject = msg.subject
        reply = msg.reply
        data = msg.data.decode()
        print("Received a message on '{subject} {reply}': {data}".format(
            subject=subject, reply=reply, data=data))

    # Simple publisher and async subscriber via coroutine.
    sub = await nc.subscribe("foo", cb=message_handler)

    # Stop receiving after 2 messages.
    await sub.unsubscribe(limit=2)
    await nc.publish("foo", b'Hello')
    await nc.publish("foo", b'World')
    await nc.publish("foo", b'!!!!!')

    # Synchronous style with iterator also supported.
    sub = await nc.subscribe("bar")
    await nc.publish("bar", b'First')
    await nc.publish("bar", b'Second')

    try:
        async for msg in sub.messages:
            print(f"Received a message on '{msg.subject} {msg.reply}': {msg.data.decode()}")
            await sub.unsubscribe()
    except Exception as e:
        pass

    async def help_request(msg):
        print(f"Received a message on '{msg.subject} {msg.reply}': {msg.data.decode()}")
        await nc.publish(msg.reply, b'I can help')

    # Use queue named 'workers' for distributing requests
    # among subscribers.
    sub = await nc.subscribe("help", "workers", help_request)

    # Send a request and expect a single response
    # and trigger timeout if not faster than 500 ms.
    try:
        response = await nc.request("help", b'help me', timeout=0.5)
        print("Received response: {message}".format(
            message=response.data.decode()))
    except TimeoutError:
        print("Request timed out")

    # Remove interest in subscription.
    await sub.unsubscribe()

    # Terminate connection to NATS.
    await nc.drain()

if __name__ == '__main__':
    asyncio.run(main())
```

As far as I understand, the output should start with the messages 
Hello 
World

But it doesn't. Here is my actual output: 
```
Received a message on 'bar ': First
Received a message on 'bar ': Second
Received a message on 'help _INBOX.5FjS6COFlT6dbkxI8O9wKS.5FjS6COFlT6dbkxI8O9wMW': help me
Received response: I can help
```
Python 3.9.9, nats-py: 2.0.0.
Am I doing something wrong?

**Comment by wallyqs** [comment]
Your expectation is right, it looks like there is an issue with the auto unsubscribe functionality since `unsubscribe(limit=2)`, should not have instantly removed the interest.

**Comment by wallyqs** [comment]
I found a fix for this and have a patch now btw, thanks again for the report.

**Comment by wallyqs** [comment]
Fixed via #280
